pmd-apex/src/test/java/net/sourceforge/pmd/lang/apex/metrics/ApexProjectMirrorTest.java

```diff
@@ -7,6 +7,7 @@
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotEquals;
 
+import com.google.summit.ast.CompilationUnit;
```

Comment:
What's the current plan for the front-end swap-ability?

If we're aiming at no swapping, run-time swaps, or checkout-time swaps (a.k.a. different git branches)... ok.

If we're aiming at compilation-time or JAR-loading-time swaps, it might be useful to establish the "boundary" of what is duplicated into a Jorje version and a Summit version.

That boundary definitely encompasses pmd/lang/apex/ast/ (right?).  What about these other files in metrics/, rule/, etc.?  If this is the target we're aiming at, would it make sense to first encapsulate these references into a new method or class that lives inside of ast/ ? (Because right now we're removing one unwanted import from Jorje and adding another unwanted import on Summit.)

Anyway, there isn't a right or wrong way... just wandering how you envision to approach this.

Reply:
My preliminary idea for swapping was to use PMD's existing mechanism for loading language modules, so the effective boundary would be at the module level (all of `pmd-apex` would be duplicated). If we go with this swapping strategy, these references outside of `ast` would also be duplicated, so removing them would have little benefit, at least relating to the swapping.

Depending on the strategy used, you are definitely right that the Jorje/Summit references could have to be contained within the `ast` package. This may be worthwhile refactoring to do regardless of the swapping (as we discussed in June in `b/236158989`); I may do this as a separate PR.